google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1113,4 +1114,289 @@ public Bucket setDefaultKmsKey(String bucketName, String kmsKeyName) throws Stor
+
+    // Select all fields
+    // Fields can be selected individually e.g. Storage.BlobField.CACHE_CONTROL
+    Blob blob = storage.get(bucketName, blobName, BlobGetOption.fields(Storage.BlobField.values()));
+
+    // Print blob metadata
+    System.out.println("Bucket: " + blob.getBucket());
+    System.out.println("CacheControl: " + blob.getCacheControl());
+    System.out.println("ComponentCount: " + blob.getComponentCount());
+    System.out.println("ContentDisposition: " + blob.getContentDisposition());
+    System.out.println("ContentEncoding: " + blob.getContentEncoding());
+    System.out.println("ContentLanguage: " + blob.getContentLanguage());
+    System.out.println("ContentType: " + blob.getContentType());
+    System.out.println("Crc32c: " + blob.getCrc32c());
+    System.out.println("ETag: " + blob.getEtag());
+    System.out.println("Generation: " + blob.getGeneration());
+    System.out.println("Id: " + blob.getBlobId());
+    System.out.println("KmsKeyName: " + blob.getKmsKeyName());
+    System.out.println("Md5Hash: " + blob.getMd5());
+    System.out.println("MediaLink: " + blob.getMediaLink());
+    System.out.println("Metageneration: " + blob.getMetageneration());
+    System.out.println("Name: " + blob.getName());
+    System.out.println("Size: " + blob.getSize());
+    System.out.println("StorageClass: " + blob.getStorageClass());
+    System.out.println("TimeCreated: " + new Date(blob.getCreateTime()));
+    System.out.println("Last Metadata Update: " + new Date(blob.getUpdateTime()));
+    System.out.println(
+        "temporaryHold: "
+            + (blob.getTemporaryHold() != null && blob.getTemporaryHold()
+                ? "Enabled"
+                : "Disabled"));
+    System.out.println(
+        "eventBasedHold: "
+            + (blob.getTemporaryHold() != null && blob.getTemporaryHold()
+                ? "Enabled"
+                : "Disabled"));
+    if (blob.getRetentionExpirationTime() != null) {
+      System.out.println("retentionExpirationTime: " + new Date(blob.getRetentionExpirationTime()));
+    }
+    if (blob.getMetadata() != null) {
+      System.out.println("User metadata:");
+      for (Map.Entry<String, String> userMetadata : blob.getMetadata().entrySet()) {
+        System.out.println(userMetadata.getKey() + " " + userMetadata.getValue());
+      }
+    }
+    // [END storage_get_metadata]
+  }
+
+  /** Example of setting a retention policy on a bucket */
+  public Bucket setRetentionPolicy(String bucketName, Long retentionPeriod)
+      throws StorageException {
+    // [START storage_set_retention_policy]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The name of a bucket, e.g. "my-bucket"
+    // String bucketName = "my-bucket";
+
+    // The retetion period for objects in bucket
+    // 1 hour in seconds
+    // Long retentionPeriod = 3600L;
+
+    Bucket bucket =
+        storage.update(
```

Comment:
This bit here is kind of confusing. Are buckets immutable? does `storage.update` return a new bucket or a reference to the original bucket?

It might be more clear if you separate the builder from the update call (and if `storage.update` returns a reference to a new bucket instead of just updating the original object, consider using a different name to suggest at that behavior)

Reply:
Bucket reference didn't exist before and is being returned by `storage.update`. I'll address the naming e.g. `bucketWithRetentionPolicy`, I don't understand what you mean by updating the original object.

Reply:
Thanks for the feedback I'll do this instead!

Reply:
Wrong call, I need a bucket context before I can call `bucket.update` this only creates a BucketInfo. I'll make the adjustment where I can.